prov/verbs/src/verbs_ep.c

```diff
@@ -240,10 +240,16 @@ static int fi_ibv_ep_close(fid_t fid)
 
 	switch (ep->util_ep.type) {
 	case FI_EP_MSG:
+		if (ep->eq)
+			fastlock_acquire(&ep->eq->lock);
+
```

Comment:
This will cause a deadlock with XRC when a message endpoint is closed (if it was not ever full connected, like a simultaneous reject).

Reply:
To avoid the deadlock for XRC codepath it will make more sense to take the lock inside of fi_ibv_ep_xrc_close() right before the call to fi_ibv_ep_destroy_xrc_qp(), and release it before fi_ibv_ep_xrc_close() returns.